cmd/options.go

```diff
@@ -117,6 +125,19 @@ func getOptions(flags *pflag.FlagSet) (lib.Options, error) {
 		}
 	}
 
+	if flags.Lookup("execution-segment").Changed {
+		executionSegmentStr, err := flags.GetString("execution-segment")
+		if err != nil {
+			return opts, err
+		}
+		segment := new(lib.ExecutionSegment)
+		err = segment.UnmarshalText([]byte(executionSegmentStr))
```

Comment:
Maybe having a constructor like function will be appropriate in this case ?

p.s. `new(lib.ExecutionSegment)` is *2* characters more than  `&lib.ExecutionSegment{}` ;)

Reply:
It has a constructor - `NewExecutionSegment(from, to *big.Rat) (*ExecutionSegment, error)`. Here we're parsing a text value to an `ExecutionSegment`, so `UnmarshalText()` seems appropriate.

Reply:
:arrow_up: